spring-cql/src/test/java/org/springframework/cassandra/core/util/CollectionUtilsUnitTests.java

```diff
@@ -0,0 +1,143 @@
+/*
+ *  Copyright 2013-2016 the original author or authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cassandra.core.util;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.sameInstance;
+import static org.junit.Assert.assertThat;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+
+import org.junit.Test;
+
+/**
+ * The CollectionUtilsUnitTests class is a test suite of test cases testing the contract and functionality
+ * of the {@link CollectionUtils} class.
+ *
+ * @author John Blum
+ * @see org.junit.Test
+ * @see org.springframework.cassandra.core.util.CollectionUtils
+ * @since 1.5.0
+ */
+public class CollectionUtilsUnitTests {
+
+	<T> List<T> asList(T... array) {
+		return Arrays.asList(array);
+	}
+
+	void assertNonNullEmptyArray(Object[] array) {
+		assertThat(array, is(notNullValue()));
+		assertThat(array.length, is(equalTo(0)));
+	}
+
+	void assertNonNullEmptyCollection(Collection<?> collection) {
+		assertThat(collection, is(notNullValue()));
+		assertThat(collection.isEmpty(), is(true));
+	}
+
+	<T> Iterable<T> newIterable(final T... elements) {
```

Comment:
Please help me understand why `newIterable` is needed. Assumed `Arrays.asList(…)` or `new ArrayList<>(…)` was sufficient for the performed tests.